src/db/spanner/models.rs

```diff
@@ -781,38 +830,50 @@ impl SpannerDb {
         &self,
         params: params::DeleteCollection,
     ) -> Result<results::DeleteCollection> {
-        let user_id = params.user_id.legacy_id as u32;
+        let fxa_kid = params.user_id.fxa_kid.clone();
+        let legacy = params.user_id.legacy_id;
         let collection_id = self.get_collection_id(&params.collection)?;
-
-        self.sql(
-            "DELETE FROM bso
-              WHERE userid = @userid
+        let mut result = self
+            .sql(
+                "DELETE FROM bso
+              WHERE fxa_kid = @fxa_kid
+                AND userid = @legacy
                 AND collection = @collectionid",
-        )?
-        .params(params! {
-            "userid" => user_id.to_string(),
-            "collectionid" => collection_id.to_string(),
-        })
-        .execute(&self.conn)?;
+            )?
+            .params(params! {
+                "fxa_kid" => fxa_kid.clone(),
+                "legacy" => legacy.to_string(),
+                "collectionid" => collection_id.to_string(),
+            })
+            .execute(&self.conn)?
+            .one_or_none()?
+            .is_some();
 
-        self.sql(
-            "DELETE FROM user_collections
-              WHERE userid = @userid
+        result = result
+            || self
+                .sql(
+                    "DELETE FROM user_collections
+              WHERE fxa_kid = @fxa_kid
+                AND userid = @legacy
                 AND collection = @collectionid",
-        )?
-        .params(params! {
-            "userid" => user_id.to_string(),
-            "collectionid" => collection_id.to_string(),
-        })
-        .execute(&self.conn)?;
-
-        self.erect_tombstone(user_id)?;
+                )?
+                .params(params! {
+                    "fxa_kid" => fxa_kid.to_string(),
+                    "legacy" => legacy.to_string(),
+                    "collectionid" => collection_id.to_string(),
+                })
+                .execute(&self.conn)?
+                .one_or_none()?
+                .is_some();
+        if result {
```

Comment:
The go version never had a check like this, but I'm +1 for adding it, it makes DELETE more idempotent (depending on your definition of it)

But you'll need to use `affected_row_count()` on SyncResult for accurate DML statement result counts. I think you get back 0 rows for such statements (so this is always false).

You can also kill the first `DELETE FROM bso` statement for simplicity here, since we have ON DELETE CASCADE for the bso table, `DELETE FROM user_collections` automatically does that delete for us

Reply:
turns out, kinda needed to make sure that spanner and mysql exhibit the same behaviors.